pylint/pyreverse/diagrams.py

```diff
@@ -189,13 +189,26 @@ def extract_relationships(self):
                 node.locals_type.items()
             ):
                 for value in values:
+                    association_type = "association"
                     if value is astroid.Uninferable:
                         continue
+
+                    # Handle multiplicity link
+                    if isinstance(value, astroid.nodes.node_classes.Subscript):
+                        children = list(value.get_children())
+                        if len(children) == 2 and children[0].name == "list":
```

Comment:
This would currently only work if something is explicitly type hinted with `list[<Type>]`. 
Multiplicity associations would also make sense for other iterable types. 
I have to check for myself what possibilities we have with ``astroid`` to generalise this without having to maintain a handmade list of possible types.